shared/gh/js/controllers/gh.api.util.js

```diff
@@ -139,25 +137,104 @@ define(['exports', 'moment', 'bootstrap-notify'], function(exports, moment) {
         }
 
-        // The number of milliseconds in one week
-        var ONE_WEEK = 1000 * 60 * 60 * 24 * 7;
-        // Convert the term start and end date to milliseconds
-        var termStartDate = new Date(term.start).getTime();
-        var termEndDate = new Date(term.end).getTime();
-        // Calculate the time difference
-        var timeDifference = Math.abs(termEndDate - termStartDate);
-        // Convert to weeks and return
-        return Math.floor(timeDifference / ONE_WEEK) + 1;
+        // Get the configuration
+        var config = require('gh.core').config;
+        // Get the correct terms associated to the current application
+        var terms = config.terms[config.academicYear];
+
+        // Retrieve the corresponding term of the specified date
+        var currentTerm = getTerm(date);
+        if (!currentTerm) {
+            return 0;
+        }
+
+        // Get the start date of the corresponding term
+        var startDate = convertISODatetoUnixDate(moment(currentTerm.start).utc().format());
+
+        // Since the terms start on a Tuesday, we have an offset of 2 days.
+        var dayOffset = ((1 / 7) * 2) - 0.01;
+
+        // Calculate and return the current academic week number
+        return Math.ceil(((date - startDate) / PERIODS['week']) - (dayOffset)) + 1;
+    };
+
+    /**
+     * Return the current term if the current date is within a term
+     *
+     * @param  {Number}    date     The date in a UNIX time format
+     * @return {Object}             The term that has the specified date in its range
+     */
+    var getTerm = exports.getTerm = function(date) {
+        if (!_.isNumber(date)) {
+            throw new Error('A valid date should be provided');
+        }
+
+        // Get the configuration
+        var config = require('gh.core').config;
+        // Get the correct terms associated to the current application
+        var terms = config.terms[config.academicYear];
+
+        // Get the current term and return its name
+        return _.find(terms, function(term) {
+
+            // Convert the dates from ISO to UNIX for easier calculation
+            var startDate = convertISODatetoUnixDate(moment(term.start).utc().format());
+            var endDate = convertISODatetoUnixDate(moment(term.end).utc().format());
+
+            // We consider weeks that only count 2 days as full academic weeks.
+            // E.g. A term starts on Tuesday 13th February, but the academic weeks always start on a Thursday.
+            //      This means that the first academic week of that term starts on the 13th and ends on the 14th.
+            //      The second academic week will start on Thursday 15th February.
+            var datePlus = convertISODatetoUnixDate(moment(date).add({'days': 6}).utc().format());
+            /* istanbul ignore else */
+            if (date < startDate && datePlus >= startDate) {
+                date = datePlus;
+            }
```

Comment:
We need this for the calendar plugin.

For example: if you navigate to the week of 08/01 - 14/01 in the week view, we want to display the displayed week as **Week 1**, since the academic week starts on Tuesday 13/01. Therefore, we need to check if the chosen date is within the range of an academic week.

![screen shot 2015-02-18 at 17 03 49](https://cloud.githubusercontent.com/assets/2194396/6252154/274e1d12-b790-11e4-8b74-c0daac0f44f2.png)